bucket_test.go

```diff
@@ -908,6 +908,49 @@ func TestBucket(t *testing.T) {
 						}
 					}
 				})
+
+				if !strings.Contains(impl.name, "S3BucketChecksums") {
+					return
+				}
+
+				t.Run("PullWithCache", func(t *testing.T) {
+					prefix := testutil.NewUUID()
+					localPath := filepath.Join(tempdir, prefix)
+
+					require.NoError(t, os.MkdirAll(localPath, 0700))
+
+					f, err := os.CreateTemp(localPath, "pull-with-cache")
+					assert.NoError(t, err)
+
+					initialInfo, err := f.Stat()
+					assert.NoError(t, err)
+
+					_, err = f.Write([]byte("test-content"))
```

Comment:
You need to stat the file after you write the content, since writing the file modifies the content. This test fails when I run it locally (I'm not sure why it doesn't fail in CI?)

Reply:
My best guess is that it doesn't fail in CI because it manages to be faster than the granularity of the modtime.

Reply:
Thanks for catching this one. I had some other stuff locally and I didn't rearrange correctly when committing it seems. The test also passed locally for me despite clearly being incorrect like this - odd.

Reply:
I am going to change this to use io.Copy as well to ensure there was no funny business with partial writes.